DevUI WorkspaceProcessor fails when files change during the processing

### Describe the bug

```
2025-09-26 11:40:48,304 ERROR [io.qua.dep.dev.IsolatedDevModeMain] (main) Failed to start quarkus: java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	[error]: Build step io.quarkus.devui.deployment.menu.WorkspaceProcessor#locateWorkspaceItems threw an exception: java.io.UncheckedIOException: java.nio.file.NoSuchFileException: /Users/adamevin/workspace/redhat/quarkus/code.quarkus.io/base/target/web-bundler/dev/lib/ddd
	at io.quarkus.devui.deployment.menu.WorkspaceProcessor.locateWorkspaceItems(WorkspaceProcessor.java:113)
	at java.base/java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:733)
	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:874)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:255)
	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
	at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2651)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2630)
	at org.jboss.threads.EnhancedQueueExecutor.runThreadBody(EnhancedQueueExecutor.java:1622)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1589)
	at java.base/java.lang.Thread.run(Thread.java:1583)
	at org.jboss.threads.JBossThread.run(JBossThread.java:501)
Caused by: java.nio.file.NoSuchFileException: /Users/adamevin/workspace/redhat/quarkus/code.quarkus.io/base/target/web-bundler/dev/lib/ddd
	at java.base/sun.nio.fs.UnixException.translateToIOException(UnixException.java:92)
	at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:106)
	at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:111)
	at java.base/sun.nio.fs.UnixFileAttributeViews$Basic.readAttributes(UnixFileAttributeViews.java:55)
	at java.base/sun.nio.fs.UnixFileSystemProvider.readAttributes(UnixFileSystemProvider.java:171)
	at java.base/java.nio.file.Files.readAttributes(Files.java:1854)
	at java.base/java.nio.file.FileTreeWalker.getAttributes(FileTreeWalker.java:220)
	at java.base/java.nio.file.FileTreeWalker.visit(FileTreeWalker.java:277)
	at java.base/java.nio.file.FileTreeWalker.next(FileTreeWalker.java:374)
	at java.base/java.nio.file.Files.walkFileTree(Files.java:2833)
	at java.base/java.nio.file.Files.walkFileTree(Files.java:2870)
	at io.quarkus.devui.deployment.menu.WorkspaceProcessor.locateWorkspaceItems(WorkspaceProcessor.java:75)
	... 10 more

	at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:376)
	at io.quarkus.runner.bootstrap.AugmentActionImpl.createInitialRuntimeApplication(AugmentActionImpl.java:293)
	at io.quarkus.runner.bootstrap.AugmentActionImpl.createInitialRuntimeApplication(AugmentActionImpl.java:61)
	at io.quarkus.deployment.dev.IsolatedDevModeMain.firstStart(IsolatedDevModeMain.java:89)
	at io.quarkus.deployment.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:432)
	at io.quarkus.deployment.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:55)
	at io.quarkus.bootstrap.app.CuratedApplication.runInCl(CuratedApplication.java:143)
	at io.quarkus.bootstrap.app.CuratedApplication.runInAugmentClassLoader(CuratedApplication.java:98)
	at io.quarkus.deployment.dev.DevModeMain.start(DevModeMain.java:104)
	at io.quarkus.deployment.dev.DevModeMain.main(DevModeMain.java:67)
Caused by: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	[error]: Build step io.quarkus.devui.deployment.menu.WorkspaceProcessor#locateWorkspaceItems threw an
```

### Expected behavior

No error

### Actual behavior

_No response_

### How to Reproduce?

This is a race condition, hard to reproduce but it happens to me a few times already

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### Quarkus version or git rev

3.28.1

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

--- a/extensions/devui/deployment/src/main/java/io/quarkus/devui/deployment/menu/WorkspaceProcessor.java
+++ b/extensions/devui/deployment/src/main/java/io/quarkus/devui/deployment/menu/WorkspaceProcessor.java
@@ -21,6 +21,8 @@ import java.util.concurrent.CompletionStage;
 import java.util.regex.Pattern;
 import java.util.stream.Collectors;
 
+import org.jboss.logging.Logger;
+
 import io.quarkus.assistant.runtime.dev.Assistant;
 import io.quarkus.deployment.Capabilities;
 import io.quarkus.deployment.Capability;
@@ -49,6 +51,7 @@ import io.quarkus.devui.spi.workspace.WorkspaceBuildItem;
  */
 @BuildSteps(onlyIf = IsLocalDevelopment.class)
 public class WorkspaceProcessor {
+    private static final Logger LOG = Logger.getLogger(WorkspaceProcessor.class);
 
     @BuildStep
     void locateWorkspaceItems(BuildSystemTargetBuildItem buildSystemTarget,
@@ -74,24 +77,34 @@ public class WorkspaceProcessor {
             try {
                 Files.walkFileTree(projectRoot, new SimpleFileVisitor<Path>() {
                     @Override
-                    public FileVisitResult preVisitDirectory(Path dir, BasicFileAttributes attrs) throws IOException {
-                        if (Files.isHidden(dir) || ignoreFolders.contains(dir.getFileName().toString())
-                                || !Files.isReadable(dir) || !Files.isExecutable(dir)) {
-                            return FileVisitResult.SKIP_SUBTREE;
+                    public FileVisitResult preVisitDirectory(Path dir, BasicFileAttributes attrs) {
+                        try {
+                            if (Files.isHidden(dir) || ignoreFolders.contains(dir.getFileName().toString())
+                                    || !Files.isReadable(dir) || !Files.isExecutable(dir)) {
+                                return FileVisitResult.SKIP_SUBTREE;
+                            }
+                        } catch (Throwable t) {
+                            // Don't let this stop startup
+                            LOG.debug("Error while getting workspace items", t);
                         }
                         return FileVisitResult.CONTINUE;
                     }
 
                     @Override
-                    public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
+                    public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) {
                         String fileName = file.getFileName().toString();
-                        boolean shouldIgnore = Files.isHidden(file)
-                                || file.startsWith(outputDir) || !Files.isReadable(file)
-                                || ignoreFilePatterns.stream().anyMatch(p -> p.matcher(fileName).matches());
+                        try {
+                            boolean shouldIgnore = Files.isHidden(file)
+                                    || file.startsWith(outputDir) || !Files.isReadable(file)
+                                    || ignoreFilePatterns.stream().anyMatch(p -> p.matcher(fileName).matches());
 
-                        if (!shouldIgnore) {
-                            String name = projectRoot.relativize(file).toString();
-                            workspaceItems.add(new WorkspaceBuildItem.WorkspaceItem(name, file));
+                            if (!shouldIgnore) {
+                                String name = projectRoot.relativize(file).toString();
+                                workspaceItems.add(new WorkspaceBuildItem.WorkspaceItem(name, file));
+                            }
+                        } catch (Throwable t) {
+                            // Don't let this stop startup
+                            LOG.debug("Error while getting workspace items", t);
                         }
                         return FileVisitResult.CONTINUE;
                     }
@@ -110,7 +123,8 @@ public class WorkspaceProcessor {
                 });
 
             } catch (IOException e) {
-                throw new UncheckedIOException(e);
+                // Don't let this stop startup
+                LOG.debug("Error while getting workspace items", e);
             }
 
             sortWorkspaceItems(workspaceItems);

